src/classes/class/Class.js

```diff
@@ -0,0 +1,35 @@
+import {useMemo} from "react";
+import {useParams} from "react-router-dom";
+import CLASSES from "../example";
+import PersonCard from "./PersonCard";
+import useUsers from "../../hooks/useUsers";
+import H1 from "../../components/H1";
+
+function Class() {
+  const {classURL} = useParams();
+
+  // if CLASSES was large, then this would be expensive so useMemo would be useful
+  const classData = useMemo(() => {
+    return CLASSES.find(c => c.url === classURL)
+  }, [classURL]);
```

Comment:
from what i can tell, the only thing that could trigger a rerender of `Class` is if `classURL` changes. (or if `users` or `message` change, but that will only happen once during the initial render). 
this means that this `useMemo` doesn't really do anything because every time `Class` rerenders, it will be because `classURL` changes, and then we would run the `useMemo` computation here. it's not saving us any work.
if, however, `Class` was dealing with other states unrelated to the classURL, and those states might be frequently changing, then `useMemo` would make sense here.

basically, the point of `useMemo` is what is *excluded* from the dependency array -- we are trying to avoid rerendering when those excluced items change -- which means there have to be reactive variables that change a lot that are *excluded* from the dependency array for `useMemo` to have any effect.